src/samples/tester.cpp

```diff
@@ -0,0 +1,232 @@
+    std::cout << entry.nickname << ": " << entry.message << std::endl; ;
+    }
+
+    std::cout << "Member A should receive it, too:" << std::endl;
+    while(chat_entries.size() == 0) {
+        chat_entries = member_a->PopChatEntries();
+        sleep(10);
+    }
+    for (const auto& entry: chat_entries) {
+    std::cout << entry.nickname << ": " << entry.message << std::endl;
+    }
+*/
+
+    //Test: Beacons
+    std::cout << std::endl;
+    std::cout << "Testing beacon tunneling" << std::endl << std::endl;
+    
+    WifiPacket beacon_packet{};
+    beacon_packet.channel=11;
+    beacon_packet.type=WifiPacket::PacketType::Beacon;
+    beacon_packet.destination_address = NoPreferredMac;
+    // packet.transmitter_address can't be set, because member->GetMacAddress is missing
+    beacon_packet.data = std::vector<uint8_t> { 1, 2, 3, 4, 5, 6, 7};
+    member_a->SendWifiPacket(beacon_packet);
+
+    std::deque<WifiPacket> packets;
+    while(packets.size() == 0)
+        packets = member_b->PopWifiPackets(WifiPacket::PacketType::Beacon);
+    for (const auto& received_packet: packets) {
+        std::cout << "Member B received beacon:" <<std::endl << "Data: ";
+        for (const auto& value: received_packet.data) {
+            std::cout << +value;
+        }
+        std::cout << std::endl;
+    }
+    std::cout << "There shouldn't be any more beacons now" << std::endl;
+    std::cout << "Member A has " << member_a->PopWifiPackets(WifiPacket::PacketType::Beacon).size() << " beacons" << std::endl;
+    std::cout << "Member B has " << member_b->PopWifiPackets(WifiPacket::PacketType::Beacon).size() << " beacons" << std::endl;
+
+    //Test: Data
+    std::cout << std::endl;
+    std::cout << "Testing data tunneling" << std::endl << std::endl;
+
+    packets.clear();
+    WifiPacket data_packet{};
+    data_packet.channel=11;
+    data_packet.type=WifiPacket::PacketType::Data;
+    data_packet.destination_address = NoPreferredMac;
+    // packet.transmitter_address can't be set, because member->GetMacAddress is missing
+    data_packet.data = std::vector<uint8_t> { 1, 2, 3, 4, 5, 6, 7};
+    member_b->SendWifiPacket(data_packet);
+
+    while(packets.size() == 0)
+        packets = member_a->PopWifiPackets(WifiPacket::PacketType::Data);
+    for (const auto& received_packet: packets) {
+        std::cout << "Member A received data:" <<std::endl << "Data: ";
+        for (const auto& value: received_packet.data) {
+            std::cout << +value;
+        }
+        std::cout << std::endl;
+    }
+    std::cout << "There shouldn't be any more data frames now" << std::endl;
+    std::cout << "Member A has " << member_a->PopWifiPackets(WifiPacket::PacketType::Data).size() << " data frames" << std::endl;
+    std::cout << "Member B has " << member_b->PopWifiPackets(WifiPacket::PacketType::Data).size() << " data frames" << std::endl;
+
+    member_b->Leave();
+    member_a->Leave();
+    room->Destroy();
+}
```

Comment:
`return 0;` ?

Maybe also add some output that the test has completed successfully